components/serviceMap/customNodes.tsx

```diff
@@ -0,0 +1,441 @@
+  const producer = OperationType[data.audience.operationType] ==
+    OperationType[OperationType.PRODUCER];
+
+  const handleModalOpen = () => {
+    setIsOpen(true);
+  };
+
+  const height = 132 + ((data.groupCount - 1) * 64);
+
+  return (
+    <div
+      class={`rounded-lg shadow-lg border-1 border-purple-200 min-w-[280px] min-h-[${height}px]`}
+    >
+      {/*<NodeResizeControl minWidth={100} minHeight={50}>*/}
+      {/*  <IconResize class="w-6 h-6" />*/}
+      {/*</NodeResizeControl>*/}
+      <div
+        id="dragHandle"
+        class="flex flex-row items-center my-2"
+      >
+        <IconGripVertical
+          class="w-6 h-6 ml-2 text-purple-100 cursor-grab bg-white border border-purple-200"
+          id="dragHandle"
+        />
+        <div class="ml-2">{data.label}</div>
+      </div>
+
+      <Handle
+        type="source"
+        position={producer ? Position.Bottom : Position.Top}
+        style={{ opacity: 0 }}
+      />
+      <Handle
+        type="target"
+        position={producer ? Position.Top : Position.Bottom}
+        style={{ opacity: 0 }}
+      />
+    </div>
+  );
+};
+
+export const OperationNode = ({ data }: { data: NodeData }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const producer = OperationType[data.audience.operationType] ==
+    OperationType[OperationType.PRODUCER];
+
+  const handleModalOpen = () => {
+    setIsOpen(true);
+  };
+
+  return (
+    <div className="h-[96px] flex flex-row justify-start items-center">
+      <div
+        type="button"
+        onClick={handleModalOpen}
+        class="flex items-center justify-betweenw-[250px] h-[64px] bg-white rounded-lg shadow-lg border-1 border-purple-200 pl-1 pr-2"
+      >
+        <div class="flex flex-row items-center">
+          <IconGripVertical
+            class="w-6 h-6 text-purple-100 cursor-grab mr-1"
+            id="dragHandle"
+          />
+
+          {producer
+            ? <ProducerIcon class="w-5 h-5 mr-2" />
+            : <ConsumerIcon class="w-5 h-5 mr-2" />}
+        </div>
+        <div class="w-[145px] whitespace-nowrap text-ellipsis overflow-hidden">
+          <a
+            href={`/service/${
+              encodeURIComponent(data.audience.serviceName)
+            }/component/${encodeURIComponent(data.audience.componentName)}/${
+              OperationType[data.audience.operationType]
+            }/op/${encodeURIComponent(data.audience.operationName)}`}
+          >
+            <div
+              class={"flex flex-col justify-start p-1"}
+            >
+              <h2
+                class={"text-[16px] whitespace-nowrap text-ellipsis overflow-hidden"}
+              >
+                {data.label}
+              </h2>
+              <h3 class="text-xs text-gray-500">
+                {titleCase(OperationType[data.audience.operationType])}
+              </h3>
+            </div>
+          </a>
+        </div>
+        <NodeMenu audience={data.audience} />
```

Comment:
We continue to have z-index issues in here. I worked around them a bit...but managing that with non-react flowbite is a real pain. I can try again to get flowbite react working...I can't remember now why it didn't work...or we can ditch the flowbite for proper react components where z-index issues continue to be a problem.